src/stream/src/executor/backfill/no_shuffle_backfill.rs

```diff
@@ -611,6 +623,26 @@ where
         }
     }
 
+    #[try_stream(ok = Option<OwnedRow>, error = StreamExecutorError)]
+    async fn make_snapshot_stream(
```

Comment:
Can we extract this part?

Reply:
Leaving it as-is for now. If there's more usage, we can refactor it into a macro.
This is because we want to instantiate the snapshot stream lazily, only if `paused` is `false`.

As such we need to pass a boxed closure (which makes the code complicated) OR use a macro (I prefer this).

But since we only have 2 call sites, we can just leave it as-is.